src/charm.py

```diff
@@ -6,56 +6,43 @@
-from ops.charm import CharmBase
-from ops.framework import StoredState
-from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus
+import ops
+from charms.filesystem_client.v0.filesystem_info import FilesystemProvides, NfsInfo
 
 logger = logging.getLogger(__name__)
 
 
-class NFSServerProxyCharm(CharmBase):
+class NFSServerProxyCharm(ops.CharmBase):
     """NFS server proxy charmed operator."""
 
-    _stored = StoredState()
-
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self._stored.set_default(endpoint=None)
-        self._nfs_share = NFSProvides(self, "nfs-share")
+        self._filesystem = FilesystemProvides(self, "filesystem", "server-peers")
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self._nfs_share.on.share_requested, self._on_share_requested)
 
     def _on_config_changed(self, _) -> None:
         """Handle updates to NFS server proxy configuration."""
-        if (endpoint := self.config.get("endpoint")) is None:
-            self.unit.status = BlockedStatus("No configured endpoint")
+        if (hostname := self.config.get("hostname")) is None:
+            self.unit.status = ops.BlockedStatus("No configured hostname.")
+            return
+
+        if (path := self.config.get("path")) is None:
+            self.unit.status = ops.BlockedStatus("No configured path.")
             return
-        elif endpoint is not None and self._stored.endpoint is None:
-            logger.debug(f"Setting NFS share endpoint to {endpoint}")
-            self._stored.endpoint = endpoint
-        else:
-            logger.warning(f"Endpoint can only be set once. Ignoring {endpoint}")
-
-        self.unit.status = ActiveStatus("Exporting share")
-
-    def _on_share_requested(self, event: ShareRequestedEvent) -> None:
-        """Handle when NFS client requests an NFS share."""
-        logger.debug(
-            (
-                "NFS share requested with parameters "
-                f"('name'={event.name}, 'allowlist'={event.allowlist}, 'size={event.size}')"
+
+        port = self.config.get("port")
+
+        self._filesystem.set_info(
+            NfsInfo(
+                hostname=str(hostname),
+                path=str(path),
+                # Port 0 it's an invalid port, so we don't need to pass it through.
+                port=int(port) if port else None,
             )
         )
-        if self._stored.endpoint is None:
-            logger.warning("Deferring ShareRequested event because endpoint is not set")
-            self.unit.status = BlockedStatus("No configured endpoint")
-            event.defer()
-        else:
-            logger.debug(f"Setting {self._stored.endpoint} as the NFS share endpoint")
-            self._nfs_share.set_endpoint(event.relation.id, endpoint=self._stored.endpoint)
+
+        self.unit.status = ops.ActiveStatus("Exporting share.")
```

Comment:
I'd leave the active status message blank to reduce the "visual clutter" in the output of `juju status` if everything is "okie dokie" with the operator:

```suggestion
        self.unit.status = ops.ActiveStatus("")
```

`active` & `idle` in `juju status` show that everything is working correctly such as exporting the NFS share.